cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -1668,8 +1678,9 @@ public List<ConfigMap> generatePerBrokerConfigurationConfigMaps(MetricsAndLoggin
                 data.put(BROKER_LISTENERS_FILENAME, listeners.stream().map(ListenersUtils::envVarIdentifier).collect(Collectors.joining(" ")));
 
                 if (useKRaft) {
-                    // In KRaft, we need to pass the Kafka CLuster ID
+                    // In KRaft, we need to pass the Kafka CLuster ID and the metadata version
                     data.put(BROKER_CLUSTER_ID_FILENAME, clusterId);
+                    data.put(BROKER_METADATA_VERSION_FILENAME, metadataVersion);
```

Comment:
wondering what's going to happen when, in my migration PR, we are going to set this configuration field on a broker which is migrating so it has both ZooKeeper and KRaft configured. I mean ... if the Kafka code will complain.

Reply:
Please see the reply to the comment about the `kafka_run.sh`.